icu4c/source/i18n/plurrule.cpp

```diff
@@ -86,54 +88,69 @@ SharedPluralRules::~SharedPluralRules() {
 
 PluralRules*
 PluralRules::clone() const {
-    return new PluralRules(*this);
+    PluralRules* newObj = new PluralRules(*this);
+    // Since clone doesn't have a 'status' parameter, the best we can do is return nullptr if
+    // the newly created object was not fully constructed properly (an error occurred).
+    if (newObj != nullptr && U_FAILURE(newObj->mInternalStatus)) {
+        delete newObj;
+        newObj = nullptr;
+    }
+    return newObj;
 }
 
 PluralRules&
 PluralRules::operator=(const PluralRules& other) {
     if (this != &other) {
         delete mRules;
-        if (other.mRules==NULL) {
-            mRules = NULL;
+        mInternalStatus = other.mInternalStatus;
```

Comment:
need to set mRules = nullptr here, otherwise the failure case could leave it with the deleted value, and the destructor could end up deleting again.

Reply:
Good point! Thank you very much for the review!
This would also somewhat simplify the if statement below. 
I will push an update shortly.